proxy/http/Http1ClientSession.cc

```diff
@@ -228,12 +225,13 @@ Http1ClientSession::do_io_close(int alerrno)
     slave_ka_vio = nullptr;
   }
   // Completed the last transaction.  Just shutdown already
-  if (transact_count == released_transactions) {
+  // Or the do_io_close is due to a network error
+  if (transact_count == released_transactions || alerrno == HTTP_ERRNO) {
     half_close = false;
   }
 
   // Clean up the write VIO in case of inactivity timeout
-  this->do_io_write(nullptr, 0, nullptr);
+  this->do_io_write(this, 0, nullptr);
```

Comment:
Does this change need to be made on this PR with other changes?

The change itself is probably fine. I think I read the benefit to not set nullptr on another PR. However, I don't think this is something to prevent crashes.

Reply:
I read the description, and now I think this is the fix, but then what are the other changes for?